File: ruby/src/least_recently_used_cache_item.rb
# frozen_string_literal: true

# Represents a single item in the LRU cache, stored as a node in a doubly-linked list.
# Each item contains a key, value, and references to the next and previous items in the list.
class LeastRecentlyUsedCacheItem
  attr_reader :key, :value
  attr_accessor :next, :prev

  def initialize(key, value)
    @key = key
    @value = value
    @next = nil
    @prev = nil
  end
end

File: typescript/src/LeastRecentlyUsedCacheItem.ts
export class LeastRecentlyUsedCacheItem<T> {
  key: string;
  value: T;
  prev: LeastRecentlyUsedCacheItem<T> | null;
  next: LeastRecentlyUsedCacheItem<T> | null;

  constructor(key: string, value: T) {
    this.key = key;
    this.value = value;
    this.prev = null;
    this.next = null;
  }
}
